File: Arrays_101/Merge Sorted Array/solution.cpp
// class Solution {
// public:
//     void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        
//         int ans[m+n];
//         int i=0,j=0,ind=0;
        
//         while(i<m && j<n)
//         {
//             if(nums1[i]<nums2[j])
//             {
//                 ans[ind] = nums1[i];
//                 i++;
//                 ind++;
//             }
//             else
//             {
//                 ans[ind] = nums2[j];
//                 j++;
//                 ind++;
//             }
//         }
        
//         while(i<m)
//         {
//             ans[ind] = nums1[i];
//             i++;
//             ind++;
//         }
        
//         while(j<n)
//         {
//             ans[ind] = nums2[j];
//             j++;
//             ind++;
//         }
        
//         for(int i=0; i<(m+n); i++)
//             nums1[i] = ans[i];
//     }
// };

// OR

class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
    

    for(int i = 0; i< n ; i++)
        nums1[m++] = nums2[i];

    sort(nums1.begin(), nums1.end());
    }

};

File: Arrays_101/Merge Sorted Array/solution.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function (nums1, m, nums2, n) {
  // nums1.splice(m,n,...nums2);
  // nums1.sort((a,b)=>a-b)

  let combined = m + n - 1,
    first = m - 1,
    second = n - 1;

  while (second >= 0) {
    if (nums1[first] > nums2[second]) {
      nums1[combined] = nums1[first];
      first--;
    } else {
      nums1[combined] = nums2[second];
      second--;
    }
    combined--;
  }
};
